src/components/scripts/oauth.mjs

```diff
@@ -0,0 +1,221 @@
+/**
```

Comment:
Please update this file from a `.mjs` file to a `.ts` file to align with the other files in this application, and provide basic typing

Reply:
Please also run `pnpm format` to align with this repo's formatting preferences.